qiskit_experiments/library/tomography/basis/local_basis.py

```diff
@@ -0,0 +1,414 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Circuit basis for tomography preparation and measurement circuits
+"""
+import functools
+from typing import Sequence, Optional, Tuple, Union, List, Dict
+import numpy as np
+from qiskit.circuit import QuantumCircuit, Instruction
+from qiskit.quantum_info import DensityMatrix, Statevector, Operator, SuperOp
+from qiskit.quantum_info.operators.channel.quantum_channel import QuantumChannel
+from qiskit.exceptions import QiskitError
+from .base_basis import PreparationBasis, MeasurementBasis
+
+# Typing object for POVM args of measurement basis
+POVM = Union[List[Statevector], List[DensityMatrix], QuantumChannel]
+
+
+class LocalPreparationBasis(PreparationBasis):
+    """Local tensor-product preparation basis.
+
+    This basis consists of a set of 1-qubit instructions which
+    are used to define a tensor-product basis on N-qubits.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        instructions: Optional[Sequence[Instruction]] = None,
+        default_states: Optional[Sequence[Union[Statevector, DensityMatrix]]] = None,
+        qubit_states: Optional[Dict[int, Sequence[Union[Statevector, DensityMatrix]]]] = None,
```

Comment:
Seems like this is new feature but test is completely missing. I cannot follow the entire logic. Do you think it's worth adding following test?
- Run QPT with standard Pauli basis.
- Create basis instances directly from `Local*Basis` by specifying only `qubit_states` of corresponding physical qubits (the state should be equal to Pauli above) and run QPT.
- Compare reconstructed channels.